walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -416,5 +416,35 @@ public void onlyBusyOptionalAttendees() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void overlappingMandatoryAndOptionalTimes() {
+    // The available times for mandatory and optional attendees overlap. Only the overlapping time
+    // should be returned.
+    //
+    // Event   : |---A---|     |---A---|
+    // Optional: |----B----|     |--B--|
+    // Day     : |---------------------|
+    // Options :           |---|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 3", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 4", TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, false),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM,
+        DURATION_30_MINUTES));
+
+    Assert.assertEquals(expected, actual);
+  }
 }
```

Comment:
Is this an extra test from the previous PR? No need to modify this since I did recommend that tests and implementation should be in the same PR.  Just an FYI

Reply:
Yep, just an extra test that I came up with. And thanks for the info!